### ISSUE TITLE
about preprocessing

### ISSUE THREAD

**Issue by JShan1001** [CHEN-XDU/GMSS#2]
Could you tell me how you preprocessed the original SEED file into npy format? and could you please tell me what your data shape is after data preprocessing? Looking forward to your reply. Thank you！

**Comment by haojiubujian91** [comment]
I have the same problem as you
Looking forward to the author's reply

**Comment by Hesper-i** [comment]
I have the same problem as you. Anybody solved it?

**Comment by hugge0620** [comment]
I have the same problem as you. Looking forward to the author's reply

**Comment by wiltontsang** [comment]
@JShan1001 
Hello

Have you solved this problem? I also have the same doubt. If it is solved, could you please tell me? 

Thank you！

**Comment by doo1o** [comment]
I have the same problem as you

**Comment by jiangchuanlu** [comment]
I have the same problem

**Comment by Thepaintedveil** [comment]
+1

**Comment by Olivia026** [comment]
@JShan1001 
Have you solved this problem? I also have the same doubt. If it is solved, could you please tell me?

Thank you very much！

**Comment by Xiaochi111** [comment]
Has anyone solved the shape problem after data preprocessing? If so, can you tell me? Thank you

**Comment by Alam-100** [comment]
I have the same problem about how to load data and preprocess the original seed dataset. Whethor the "path":'/GNN_NPY_DATASETS/SEED/data_dependent'  is the pre-data's path or not. It's very important for us to known! Thank for your reply!

**Comment by lr580** [comment]
Based on author's codes and the description of datasets, I wrote a data preprocessing script for unsupervised mode subject-dependent experiment and single SEED dataset, here are the codes [link](https://gist.github.com/lr580/3f280907604faa064f4bfe02f372d276)

I dont know whether standardization is needed in data preprocessing, so I skip it but added `nn.BatchNorm1d` in the GMSS's code at `class SelfSupervisedTest` (`model.py`) to keep the performance. (If standardization is added in data preprocessing, I ran very low accuracy for the jigsaw tasks). Due to time and equipment constraints, I only ran partially the experiments, and I got about 85% average accuracy across 9 runs at `SelfSuprevised.py`.

As a beginner, I acknowledge there might be potential issues in my preprocessing implementation. Nevertheless, I hope this contribution can be helpful to your working.

**Comment by Alam-100** [comment]
Thank you very much! This is very important for me! (谢谢你！本来还以为没有回复咯，你真的很贴心！我们可能还是校友呢(xidian))



---Original---
From: ***@***.***&gt;
Date: Tue, Apr 1, 2025 10:55 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [CHEN-XDU/GMSS] about preprocessing (Issue #2)


   
Based on author's codes and the description of datasets, I wrote a data preprocessing script for unsupervised mode subject-dependent experiment and single SEED dataset, here are the codes link
 
I dont know whether standardization is needed in data preprocessing, so I skip it but added nn.BatchNorm1d in the GMSS's code at class SelfSupervisedTest (model.py) to keep the performance. (If standardization is added in data preprocessing, I ran very low accuracy for the jigsaw tasks). Due to time and equipment constraints, I only ran partially the experiments, and I got about 85% average accuracy across 9 runs at SelfSuprevised.py.
 
As a beginner, I acknowledge there might be potential issues in my preprocessing implementation. Nevertheless, I hope this contribution can be helpful to your working.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  lr580 left a comment (CHEN-XDU/GMSS#2)
 
Based on author's codes and the description of datasets, I wrote a data preprocessing script for unsupervised mode subject-dependent experiment and single SEED dataset, here are the codes link
 
I dont know whether standardization is needed in data preprocessing, so I skip it but added nn.BatchNorm1d in the GMSS's code at class SelfSupervisedTest (model.py) to keep the performance. (If standardization is added in data preprocessing, I ran very low accuracy for the jigsaw tasks). Due to time and equipment constraints, I only ran partially the experiments, and I got about 85% average accuracy across 9 runs at SelfSuprevised.py.
 
As a beginner, I acknowledge there might be potential issues in my preprocessing implementation. Nevertheless, I hope this contribution can be helpful to your working.

**Comment by starsgo** [comment]
I have the same problem
